contributors/devel/sig-architecture/api_changes.md

```diff
@@ -1182,10 +1182,13 @@ provides some details
     }
     ```
 
+    Because validation is always on, the field can be changed from a previously invalid value to a valid value, or cleared, even if the feature gate is disabled, but the field cannot be updated to be invalid, and an invalid value will prevent other changes to the object.
+
 In future Kubernetes versions:
 
 * if the feature progresses to beta or stable status, the feature gate can be removed or be enabled by default.
 * if the schema of the alpha field must change in an incompatible way, a new field name must be used.
+* if the validation rules of the alpha field are changed across versions (including from a release where a field is alpha to a release where a field is beta), it is the user's responsibility to update their fields to be valid.
```

Comment:
I think there's another scenario: Specifically for alpha->beta, the n-1 (alpha) validation logic needs to be compatible with the n (beta) logic if a user starts using the beta feature in version n.

Reply:
But given the comment above about not expecting alpha object or fields to behave the same after an upgrade to beta, I think this is too strong? 

The n-1 alpha validation being different than the n beta validation is exactly this scenario I thought.

Reply:
I think it applies to alpha->alpha or an alpha (n-1) object ->beta.

But I am thinking specifically about a beta (n) object interacting with an alpha (n-1) server, which could happen normally in an upgrade scenario.

Reply:
I think the mindset I'm challenging "alpha on in (n-1) -> beta" is supported.  If it's not supported, and the field shouldn't be populated, then validation changing doesn't matter.  We're not saying you *can't* populate the alpha field and then upgrade to beta, but I think I'm arguing that's EXACTLY the same scenario as "alpha (n-1) to incompatible validation alpha (n)" (because alpha from is what is unsupported, not alpha to)

Reply:
This sounds OK in theory, but there are a hundred ways it could go sideways:

* the field might be immutable (we do suggest not allowing updates unless you
  know why it will be updated).

* Since we validate data on the way in, most controller code DOES NOT check
  that again.  Concretely, controller code might take a value that was valid in
  alpha but is not valid in beta and use it directly.  This could cause
  arbitrary damage to the cluster.

Again, since this doc is speaking to k8s devs, we should encourage them to
think REALLY hard about the impact of a breaking change on existing users.d
This may include adding controller code to handle old/bad values gracefully.

Reply:
IOW "it's alpha" is still not a good excuse to cause harm when it can reasonably be avoided.

Reply:
I can absolutely instruct kube developers to self-flagellate in this paragraph.  The trigger was the lack of clarity about what users should expect, but instructing developers is also important